src/shared/interceptors/snake-to-camel.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  NestInterceptor,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { mapKeys, camelCase } from 'lodash';
+
+@Injectable()
+export class SnakeToCamelInterceptor implements NestInterceptor {
+  intercept(ctx: ExecutionContext, next: CallHandler<any>): Observable<any> {
+    const request = ctx.switchToHttp().getRequest();
+
+    if (request.query) {
+      request.query = mapKeys(request.query, (value, key) => {
+        if (/[-_]{2,}|[^a-z0-9_-]/.test(key)) {
```

Comment:
What does it mean?

Reply:
It checks has query multiple separators or invalid characters.

For now, I make it more undestandable